rm-datasource/src/main/java/io/seata/rm/datasource/exec/SelectForUpdateExecutor.java

```diff
@@ -60,9 +67,17 @@ public T doExecute(Object... args) throws Throwable {
         String selectPKSQL = buildSelectSQL(paramAppenderList);
         try {
             if (originalAutoCommit) {
+                /*
+                 * prevent release local lock before get global lock
+                 * dose not need the savepoint if original auto commit was true
+                 */
                 conn.setAutoCommit(false);
+            } else if (!originalAutoCommit && dbmd.supportsSavepoints()) {
+                //need the savepoint if original auto commit was false
```

Comment:
I think logic below is enough:

```
else if (conn.getMetaData().supportsSavepoints())
```

By the way, does single line comment need a space after `//`? e.g. `// xxx`.